src/main/java/io/pravega/connectors/flink/serialization/PravegaDeserializationSchema.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *   
+ */
+package io.pravega.connectors.flink.serialization;
+
+import io.pravega.client.stream.Serializer;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.streaming.util.serialization.AbstractDeserializationSchema;
+
+/**
+ * A deserialization schema adapter for a Pravega serializer.
+ */
+public class PravegaDeserializationSchema<T extends Serializable> extends AbstractDeserializationSchema<T> {
```

Comment:
This class name comes from the sample projects. It is the same as was used there. It is not specific to pravega but it does use Pravega's Serializer interface.

The term "schema" is a flink thing. They call them Deserialization/Serialization Schemas. I don't love the term either but it does implement the "AbstractDeserializationSchema" flink class so the name does make sense. I think it is best to be consistent with Flinks terminology here.